feat: optimize file upload and optimize the transaction

szl97 opened the pull request:
# Pull Request

## Description

- use large buffer when tomcat  copies data to resolve the multipart http protocol
- use tcp keepAlive with the connection with S3 server
- forbidden md5 hash when request S3
- cancel store a temp file when upload file

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [x] 🧹 Code refactoring (no functional changes)
- [x] ⚡ Performance improvement
- [x] 🔧 Configuration change

## Component

Please check the components affected by this PR:

- [x] 🔧 API (Backend)
- [ ] 🎨 Web (Frontend)
- [ ] 🐳 Docker/Infrastructure
- [ ] 📖 Documentation
- [ ] 🧪 Tests
- [x] 📦 Dependencies

## Testing

Please describe the tests that you ran to verify your changes:

- [ ] Unit tests pass (`npm test` / `mvn test`)
- [x] Integration tests pass
- [x] Manual testing completed
- [x] Docker build successful

**Test Configuration**:
- OS: [e.g. Ubuntu 20.04]
- Node.js version: [e.g. 20.x]
- Java version: [e.g. 8]

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Screenshots (if applicable)

Please add screenshots to help explain your changes.

## Additional Notes

Add any additional notes, concerns, or questions here.

Jiakaic commented:
@claude 检查提交的代码，使用中文回复

Jiakaic commented on api/src/main/java/com/ke/bella/files/service/FileService.java:
```diff
@@ -253,8 +360,39 @@ public OpenAIFile upload(
+     * 清理失败的上传记录（删除数据库记录和闭包表记录）
+     */
+    @Transactional
+    public void deleteFileRecord(String fileId, FileType fileType) {
```
拉下最新代码看一下，现在不同purpose文件有不同存储逻辑；file_closure表仅包含user_purpose的文件和目录；

Jiakaic commented on api/src/main/java/com/ke/bella/files/service/FileService.java:
```diff
@@ -199,13 +210,64 @@ public OpenAIFile upload(
+        }
+
+        // 步骤3：查询并广播
+        return self.finalizeFileUpload(context.getFileId(), metadata);
```
步骤1的FileUploadContext可以直接return FileDB么？否则这里可能会查不到文件导致上传文件报404？

szl97 committed:
feat: optimize file upload and optimize the transaction

- use large buffer when tomcat  copies data to resolve the multipart http protocol
- use tcp keepAlive with the connection with S3 server
- forbidden md5 hash when request S3
- cancel store a temp file when upload file
